commands/scan/scan.go

```diff
@@ -208,6 +215,14 @@ func (scanCmd *ScanCommand) RunAndRecordResults(recordResFunc func(scanResults *
 	scanResults.XrayVersion = xrayVersion
 
 	scanResults.ExtendedScanResults.EntitledForJas, err = jas.IsEntitledForJas(xrayManager, xrayVersion)
+	dynamicTokenVersionMismatchErr := clientutils.ValidateMinimumVersion(clientutils.Xray, xrayVersion, jasutils.DynamicTokenValidationMinXrayVersion)
+	if dynamicTokenVersionMismatchErr != nil && (scanCmd.validateSecrets) {
+		log.Warn("Token validation (--validate-secrets flag) is not supported in your xray version")
+		scanResults.ExtendedScanResults.SecretValidation = false
+	} else {
+		scanResults.ExtendedScanResults.SecretValidation = jas.CheckForSecretValidation(xrayManager, scanCmd.validateSecrets)
+	}
```

Comment:
to reduce code duplication and to make sure this is always running, put `clientutils.ValidateMinimumVersion` inside of `CheckForSecretValidation` right before running the API call.
(also true for the one in audit)